WcaOnRails/spec/features/delegate_report_spec.rb

```diff
@@ -0,0 +1,33 @@
+# frozen_string_literal: true
+
+require "rails_helper"
+
+RSpec.feature "Registration management" do
+  let(:delegate) { FactoryGirl.create :delegate, name: "Jeremy on Bart" }
+  let(:competition) { FactoryGirl.create :competition, :registration_open, delegates: [delegate], name: "Submit Report 2017" }
+
+  context "when signed in as competition delegate" do
+    before :each do
+      sign_in delegate
+    end
+
+    scenario "view, edit, save, submit report" do
+      # View report
+      visit "/competitions/#{competition.id}/report"
+      expect(page).to have_text("Submit Report 2017")
```

Comment:
Not everyone agrees with this, and feel free to not do it, but I prefer the dryer `.to have_text(competition.name)`.

Also `"Report submitted by #{delegate.name}"` below.